libs/s25main/ingameWindows/iwAction.cpp

```diff
@@ -338,12 +338,14 @@ void iwAction::AddUpgradeRoad(ctrlGroup* group, unsigned& /*x*/, unsigned& width
     }
 }
 
-void iwAction::DoUpgradeRoad()
+bool iwAction::DoUpgradeRoad()
 {
     Direction flag_dir;
     const noFlag* flag = gwv.GetWorld().GetRoadFlag(selectedPt, flag_dir);
     if(flag)
-        GAMECLIENT.UpgradeRoad(flag->GetPos(), flag_dir);
+        return GAMECLIENT.UpgradeRoad(flag->GetPos(), flag_dir);
+    else
+        return false;
```

Comment:
**Maybe**:

```suggestion
        return true;
```

If there is no flag anymore, the task is done and the window can be closed, or what do you think?

Reply:
I think you are exactly right.